Source/Library/GameBoardWindow/GameBoardPanel.cpp

```diff
@@ -0,0 +1,45 @@
+#include <QGridLayout>
+#include "TicTacToe/Library/GameBoardWindow/GameBoardPanel.hpp"
+#include "TicTacToe/Library/GameBoardWindow/TokenTile.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace TicTacToe;
+
+GameBoardPanel::GameBoardPanel(const Board& board, QWidget* parent) :
+  QWidget(parent),
+  m_board(&board) {
+  setFixedSize(304, 304);
+  setStyleSheet("background-color: #FFFFFF;");
+  auto panel_layout = new QGridLayout();
+  panel_layout->setContentsMargins(0, 0, 0, 0);
+  panel_layout->setSpacing(2);
+  for(auto i = 0; i < Board::ROW_COUNT; ++i) {
+    for(auto j = 0; j < Board::COLUMN_COUNT; ++j) {
+      auto tile = new TokenTile();
```

Comment:
Same here, indicate the parent when constructing, you can set the parent to ```this``` and then on line 23 ownership will be transfered to ```panel_layout```.

Reply:
I found below from Qt documentation:
When you use a layout, you do not need to pass a parent when constructing the child widgets. The layout will automatically reparent the widgets (using QWidget::setParent()) so that they are children of the widget on which the layout is installed.

Even the child widgets can be reparented, do we need to specify their parent windows?

Reply:
Yep, being explicit about ownership makes it easier for me when reading code to be confident there are no memory leaks.